### ISSUE TITLE
argparse handling multiple "--" in args improperly

### ISSUE THREAD

**Issue by hpaulj** [nodeca/argparse#57]
Python Issue 13922

`ArgumentParser._getValues()` deletes all `'--'` from the argStrings if the action is not PARSER or REMAINDER.

A patch in the Python development branch changes that to delete only the 1st '--'.   However this ends up deleting only the first in each cluster of positional strings.  I have submitted a patch that deletes only one '--', the one that puts a '-' in the `argStringsPattern` near the start of `ArgumentParser._parseKnownArgs`.
http://bugs.python.org/issue13922#msg186884

This change might apply if `argparse` is used to pass a list of arguments to another program that uses `'--'` in the same way.

**Comment by ikokostya** [comment]
I have the same problem.

**Versions**

* argparse: `1.0.10`

**Example**

```js
'use strict';

const {ArgumentParser} = require('argparse');

const parser = new ArgumentParser();
parser.addArgument('-a');
parser.addArgument('-b');

const [args, rest] = parser.parseKnownArgs();
console.log(args, rest)
```

**Expected behavior** 

Stop processing arguments after `--`:

```
$ node t.js -a 1 -- -b 2
Namespace { a: '1', b: null } [ '--', '-b', '2' ]
```

Compare with `Python 2.7.1` version:

```python
from argparse import ArgumentParser

parser = ArgumentParser()
parser.add_argument('-a')
parser.add_argument('-b')

[args, rest] = parser.parse_known_args()
print(args, rest)
```

```
$ python t.py -a 1 -- -b 2
(Namespace(a='1', b=None), ['--', '-b', '2'])
```

**Actual behavior**

Arguments after `--` are considered as known.

```
$ node t.js -a 1 -- -b 2
Namespace { a: '1', b: '2' } [ '--' ]
```

**Comment by rlidwka** [comment]
fixed in argparse 2.0
